criu/flog.c

```diff
@@ -150,12 +153,30 @@ int flog_close(int fdout)
 	return 0;
 }
 
-int flog_encode_msg(int fdout, unsigned int nargs, unsigned int mask, const char *format, ...)
+int flog_encode_msg(int loglevel, unsigned int nargs, unsigned int mask, const char *format, ...)
 {
 	flog_msg_t *m;
 	va_list argptr;
 	char *str_start, *p;
 	size_t i;
+	int fdout;
+	unsigned int current_loglevel;
+
+	if (!opts.log_in_binary){
+		va_start(argptr, format);
+		vprint_on_level(loglevel, format, argptr);
+		va_end(argptr);
+		return 0;
+	}
+
+	if (unlikely(loglevel == LOG_MSG)) {
+		fdout = STDOUT_FILENO;
+	}else {
```

Comment:
I'm not sure I read it correctly: does it mean that for LOG_MSG messages we get binary output out of criu?
So that criu-check and stuff like that will break tty setup for a user?
Should those messages just get printed instead?